src/plan/mutator_context.rs

```diff
@@ -36,7 +54,38 @@ pub(crate) fn unreachable_release_func<VM: VMBinding>(
     unreachable!("`MutatorConfig::release_func` must not be called for the current plan.")
 }
 
+/// An mutator release implementation for plans that use [`crate::plan::global::CommonPlan`].
+#[allow(unused_variables)]
+pub(crate) fn common_release_func<VM: VMBinding>(mutator: &mut Mutator<VM>, _tls: VMWorkerThread) {
+    // Release the free list allocator used for non moving
+    #[cfg(feature = "marksweep_as_nonmoving")]
+    {
+        use crate::util::alloc::FreeListAllocator;
+        unsafe {
+            mutator
+                .allocators
+                .get_allocator_mut(mutator.config.allocator_mapping[AllocationSemantics::NonMoving])
+        }
+        .downcast_mut::<FreeListAllocator<VM>>()
+        .unwrap()
+        .release();
+    }
+    #[cfg(not(any(feature = "immortal_as_nonmoving", feature = "marksweep_as_nonmoving")))]
```

Comment:
We can use `cfg_if!`.  Although "immortal_as_nonmoving" needs no action in this function. I still think it is more explicit and more readable to have something like:

```rust
cfg_if::cfg_if! {
    if #[cfg(feature = "immortal_as_nonmoving")] {
        // We don't need to do anything for ImmortalSpace.
    } else if #[cfg(feature = "mark_sweep_as_nonmoving")] {
        // ...
    } else {
        // ...
    }
}
```

At least we won't miss any cases if we write this way.